samples/browser/src/main/java/org/mozilla/samples/browser/MainActivity.kt

```diff
@@ -48,6 +48,15 @@ class MainActivity : AppCompatActivity() {
         toolbarFeature?.stop()
     }
 
+    override fun onBackPressed() {
+        val handledBySession = sessionFeature?.handleBackPressed()
+        val handledByToolbar = toolbarFeature?.handleBackPressed()
+
+        if (!(handledBySession == true || handledByToolbar == true)) {
+            super.onBackPressed()
+        }
+    }
+
```

Comment:
This will need to be ordered and exit early I think:

1. Toolbar
2. Session
3. (parent) Activity

If a previous component handled the back press then it shouldn't be forwarded to the next one.

Reply:
Yes, that makes sense. I was trying to simplify in a sense that eventually we'd loop over all features and invoke their `handleBackPressed`, where order doesn't matter. If none handled the event, we'd call `super.`, but in either case, I think that will be in the apps control. So, fine with changing it here.